src/main/java/io/github/darealturtywurty/ancientology/core/util/CommonEvents.java

```diff
@@ -25,6 +39,15 @@ private CommonEvents() {
         private ForgeEvents() {
             throw new IllegalAccessError("Illegal access to hidden event bus subscriber class!");
         }
+
+        @SubscribeEvent
+        public static void removeFlightEffect(PotionEvent.PotionRemoveEvent e) {
+            if (Objects.requireNonNull(e.getPotionEffect()).getEffect().equals(MobEffectInit.FLIGHT.get())) {
+                ((Player)e.getEntityLiving()).getAbilities().flying = false;
+                ((Player)e.getEntityLiving()).getAbilities().mayfly = false;
+                ((Player)e.getEntityLiving()).getAbilities().setFlyingSpeed(0.05F);
+            }
+        }
```

Comment:
This and the previous are casts without checking. This will crash if any other entity has the effect. Please fix this.